src/vtkF3DAssimpImporter.cxx

```diff
@@ -0,0 +1,1066 @@
+
+        if (rotationKey == nodeAnim->mRotationKeys)
+        {
+          quaternion = rotationKey->mValue;
+        }
+        else if (rotationKey == nodeAnim->mRotationKeys + nodeAnim->mNumRotationKeys)
+        {
+          quaternion = (rotationKey - 1)->mValue;
+        }
+        else
+        {
+          aiQuatKey* prev = rotationKey - 1;
+          ai_real d = (timestep - prev->mTime) / (rotationKey->mTime - prev->mTime);
+          quaternionInterpolator(quaternion, *prev, *rotationKey, d);
+        }
+
+        aiVectorKey* scalingKey =
+          std::lower_bound(nodeAnim->mScalingKeys, nodeAnim->mScalingKeys + nodeAnim->mNumScalingKeys,
+            timestep, [](const aiVectorKey& key, const double& time) { return key.mTime < time; });
+
+        if (scalingKey == nodeAnim->mScalingKeys)
+        {
+          scaling = scalingKey->mValue;
+        }
+        else if (scalingKey == nodeAnim->mScalingKeys + nodeAnim->mNumScalingKeys)
+        {
+          scaling = (scalingKey - 1)->mValue;
+        }
+        else
+        {
+          aiVectorKey* prev = scalingKey - 1;
+          ai_real d = (timestep - prev->mTime) / (scalingKey->mTime - prev->mTime);
+          vectorInterpolator(scaling, *prev, *scalingKey, d);
+        }
+
+        vtkMatrix4x4* transform = this->Internals->NodeLocalMatrix[nodeAnim->mNodeName.data];
+
+        if (transform)
+        {
+          // Initialize quaternion
+          vtkQuaternion<double> rotation;
+          rotation.Set(quaternion.w, quaternion.x, quaternion.y, quaternion.z);
+          rotation.Normalize();
+
+          double rotationMatrix[3][3];
+          rotation.ToMatrix3x3(rotationMatrix);
+
+          // Apply transformations
+          for (int i = 0; i < 3; i++)
+          {
+            for (int j = 0; j < 3; j++)
+            {
+              transform->SetElement(i, j, scaling[j] * rotationMatrix[i][j]);
+            }
+            transform->SetElement(i, 3, translation[i]);
+          }
+        }
+      }
+    }
+  }
+
+  vtkNew<vtkMatrix4x4> identity;
+  this->Internals->UpdateNodeTransform(this->Internals->Scene->mRootNode, identity);
+
+  this->Internals->UpdateBones();
+  this->Internals->UpdateCameras();
+  this->Internals->UpdateLights();
+}
+
+//----------------------------------------------------------------------------
+vtkIdType vtkF3DAssimpImporter::GetNumberOfAnimations()
+{
+  return this->Internals->Scene ? this->Internals->Scene->mNumAnimations : 0;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkF3DAssimpImporter::GetAnimationName(vtkIdType animationIndex)
+{
+  return this->Internals->Scene->mAnimations[animationIndex]->mName.C_Str();
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DAssimpImporter::EnableAnimation(vtkIdType animationIndex)
+{
+  this->Internals->EnabledAnimations[animationIndex] = true;
```

Comment:
since only one animation can be enabled, should other animation indices be disabled ?

Reply:
All other animations are disabled by default

Reply:
what happens if a code do this:

```
vtkF3DAssimpImporter::EnableAnimation(0);
vtkF3DAssimpImporter::EnableAnimation(1);
``` 
?